spec/mongo/cached_cursor_spec.rb

```diff
@@ -0,0 +1,81 @@
+require 'spec_helper'
+
+describe Mongo::CachedCursor do
+
+  around do |spec|
+    Mongo::QueryCache.clear_cache
+    Mongo::QueryCache.cache { spec.run }
+  end
+
+  let(:authorized_collection) do
+    authorized_client['cached_cursor']
+  end
+
+  before do
+    authorized_collection.drop
+  end
+
+  let(:server) do
+    view.send(:server_selector).select_server(authorized_client.cluster)
+  end
+
+  let(:reply) do
+    view.send(:send_initial_query, server)
+  end
+
+  let(:cursor) do
+    described_class.new(view, reply, server)
+  end
+
+  context 'when query cache enabled documents are cached' do
+
+    let(:view) do
+      Mongo::Collection::View.new(authorized_collection)
+    end
+
+    let(:documents) do
+      (1..3).map{ |i| { field: "test#{i}" }}
+    end
+
+    before do
+      authorized_collection.insert_many(documents)
+    end
+
+    context 'when query cache enabled' do
+
+      it 'docs are cached when query cache enabled and query is repeated' do
```

Comment:
The descriptions for `it` blocks can just explain the behavior of the test (e.g. `it 'caches the result documents'`).